src/functions/keycloak/find-user/handler.ts

```diff
@@ -33,11 +24,27 @@ const findUserHandler: APIGatewayProxyHandler = async (
       throw new Error("Missing or invalid path parameter: id");
     }
 
-    const userById = await api.findUser(queryStringParameters.id);
+    const userById: User = await api.findUser(queryStringParameters.id);
+
+    if (!userById) {
+      return {
+        statusCode: 404,
+        body: JSON.stringify({ error: "User not found" }),
+      };
+    }
+    const filteredUser: User = {
+      id: userById.id,
+      firstName: userById.firstName,
+      lastName: userById.lastName,
+      email: userById.email,
+      isActive: userById.isActive,
+      severaGuid: userById.severaGuid,
+      forecastId: userById.forecastId
+    };
```

Comment:
I'm a bit confused, why are we filtering user here? Is it because the response from the api.findUser is not actually of User type? In which case the response type should be updated to reflect what is actually returned from the request.